Foundation/NSURLSession/libcurlHelpers.swift

```diff
@@ -0,0 +1,53 @@
+// Foundation/NSURLSession/libcurlHelpers - NSURLSession & libcurl
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+// -----------------------------------------------------------------------------
+///
+/// These are libcurl helpers for the NSURLSession API code.
+/// - SeeAlso: https://curl.haxx.se/libcurl/c/
+/// - SeeAlso: NSURLSession.swift
+///
+// -----------------------------------------------------------------------------
+
+
+import CoreFoundation
+import Dispatch
+
+
+//TODO: Move things in this file?
+
+
+internal func initializeLibcurl() {
+    try! CFURLSessionInit().asError()
+}
+
+
+
+
+internal extension String {
+    /// Create a string by a buffer of UTF 8 code points that is not zero
+    /// terminated.
+    init?(utf8Buffer: UnsafeBufferPointer<UInt8>) {
+        var bufferIterator = utf8Buffer.makeIterator()
+        var codec = UTF8()
+        var result: String = ""
+        iter: repeat {
```

Comment:
I am dubious of the actual perf hit of this; (not a blocking issue) I think we can probably do much better than this and funnel it to a common interface in CF